### ISSUE TITLE
freyja aggregate sending IndexError

### ISSUE THREAD

**Issue by dfeistel3** [andersen-lab/Freyja#14]
im trying to call:
`freyja aggregate --output output.tsv DEMIX/`
where `DEMIX` is the directory with output files from `freyja demix` but i keep getting an IndexError

```Traceback (most recent call last):
  File "/scicomp/home-pure/ofx5/miniconda3/envs/freyja-env2.0/bin/freyja", line 10, in <module>
    sys.exit(cli())
  File "/scicomp/home-pure/ofx5/miniconda3/envs/freyja-env2.0/lib/python3.7/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/scicomp/home-pure/ofx5/miniconda3/envs/freyja-env2.0/lib/python3.7/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/scicomp/home-pure/ofx5/miniconda3/envs/freyja-env2.0/lib/python3.7/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/scicomp/home-pure/ofx5/miniconda3/envs/freyja-env2.0/lib/python3.7/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/scicomp/home-pure/ofx5/miniconda3/envs/freyja-env2.0/lib/python3.7/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/scicomp/home-pure/ofx5/miniconda3/envs/freyja-env2.0/lib/python3.7/site-packages/freyja/_cli.py", line 100, in aggregate
    df_demix = agg(results)
  File "/scicomp/home-pure/ofx5/miniconda3/envs/freyja-env2.0/lib/python3.7/site-packages/freyja/utils.py", line 13, in agg
    df_demix.index = [x.split('/')[1] for x in df_demix.index]
  File "/scicomp/home-pure/ofx5/miniconda3/envs/freyja-env2.0/lib/python3.7/site-packages/freyja/utils.py", line 13, in <listcomp>
    df_demix.index = [x.split('/')[1] for x in df_demix.index]
IndexError: list index out of range
```

any help?

**Comment by joshuailevy** [comment]
Hi! 
This is likely a result of a slight difference in how you've structured the variant/depth files from what I've been doing, which of course should not lead to any issues. 

I think the line you reference in your error report
 ``` df_demix.index = [x.split('/')[1] for x in df_demix.index] ``` 
needs to be changed to 
 ``` df_demix.index = [x.split('/')[-1] for x in df_demix.index] ``` 
to handle more general file structures. 

I'll fix that asap. Thanks for catching this! 
Josh

**Comment by lskatz** [comment]
Hi, I might have a similar problem?  How are the files supposed to look for `aggregate`?  I have them so that each sample has its own subdirectory like so:

```
(freyja) [gzu2@monolith3 freyja.tmp]$ ls *
SRR15128768.freyja:
freyja.demix  freyja.depths  freyja.variants.tsv  ivar.tsv

SRR15128769.freyja:
freyja.demix  freyja.depths  freyja.variants.tsv  ivar.tsv

SRR15128778.freyja:
freyja.demix  freyja.depths  freyja.variants.tsv  ivar.tsv

SRR15128779.freyja:
freyja.demix  freyja.depths  freyja.variants.tsv  ivar.tsv
```

Is it possible to document how these files are supposed to look?  And how we might key into the date feature so that we get those awesome plots?

**Comment by lskatz** [comment]
I don't know if it helps to see the raw code but this is sort of where I am so far.  https://github.com/lskatz/lskScripts/blob/master/qsub/launch_freyja.sh

**Comment by joshuailevy** [comment]
It looks like this is likely due to the same error. Your 
''' mktemp --directory ``` command ends up leads you to a slightly different file structure than what I had. 

I believe I did fix this issue in the freyja 1.2.1 release (now available on bioconda). Are you using that version?

**Comment by joshuailevy** [comment]
Ah, I may have misunderstood then. For aggregate, all of the ```.demix``` files (which I assume are your outputs from the ```freyja demix``` step) in your earlier comment should be stored in a single folder. You then just have to provide that folder to the ```aggregate``` command as shown in the README. 
```
freyja aggregate [directory-of-output-files] --output [aggregated-filename.tsv]
```

Adding a ```--version``` command is a nice touch. I'll also throw that in there soon!
